src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1744,14 +1744,6 @@ public override void VisitMarkupMinimizedTagHelperDirectiveAttribute(MarkupMinim
                             out var parameterMatch,
                             out var associatedAttributeParameterDescriptor))
                         {
-                            var expectsBooleanValue = associatedAttributeDescriptor.ExpectsBooleanValue(attributeName);
-
-                            if (!expectsBooleanValue)
-                            {
-                                // We do not allow minimized non-boolean bound attributes.
-                                return;
-                            }
-
```

Comment:
@ajaybhargavb - helped me with this. We were only handling minimized attributes for the top-level directive attributes, not for the parameters.